src/__swaps__/screens/Swap/providers/swap-provider.tsx

```diff
@@ -37,24 +37,30 @@ import { loadWallet } from '@/model/wallet';
 import { walletExecuteRap } from '@/raps/execute';
 import { queryClient } from '@/react-query';
 import { userAssetsQueryKey as swapsUserAssetsQueryKey } from '@/__swaps__/screens/Swap/resources/assets/userAssets';
-import { userAssetsQueryKey } from '@/resources/assets/UserAssetsQuery';
+import { SwapAssetType, inputKeys } from '@/__swaps__/types/swap';
 import { useAccountSettings } from '@/hooks';
-import { getGasSettingsBySpeed, getSelectedGas, getSelectedGasSpeed } from '../hooks/useSelectedGas';
+import * as i18n from '@/languages';
+import { userAssetsQueryKey } from '@/resources/assets/UserAssetsQuery';
+
+import { equalWorklet, lessThanOrEqualToWorklet, toScaledIntegerWorklet } from '@/__swaps__/safe-math/SafeMath';
+import { analyticsV2 } from '@/analytics';
 import { LegacyTransactionGasParamAmounts, TransactionGasParamAmounts } from '@/entities';
-import { equalWorklet } from '@/__swaps__/safe-math/SafeMath';
-import { useSwapSettings } from '../hooks/useSwapSettings';
-import { useSwapOutputQuotesDisabled } from '../hooks/useSwapOutputQuotesDisabled';
 import { getNetworkObj } from '@/networks';
 import { userAssetsStore } from '@/state/assets/userAssets';
-import { analyticsV2 } from '@/analytics';
 import { Address } from 'viem';
+import { getGasSettingsBySpeed, getSelectedGas, getSelectedGasSpeed } from '../hooks/useSelectedGas';
+import { useSwapOutputQuotesDisabled } from '../hooks/useSwapOutputQuotesDisabled';
+import { SyncGasStateToSharedValues, SyncQuoteSharedValuesToState } from './SyncSwapStateAndSharedValues';
 
 const swapping = i18n.t(i18n.l.swap.actions.swapping);
 const tapToSwap = i18n.t(i18n.l.swap.actions.tap_to_swap);
 const save = i18n.t(i18n.l.swap.actions.save);
 const enterAmount = i18n.t(i18n.l.swap.actions.enter_amount);
 const review = i18n.t(i18n.l.swap.actions.review);
 const fetchingPrices = i18n.t(i18n.l.swap.actions.fetching_prices);
+const selectToken = i18n.t(i18n.l.swap.actions.select_token);
+const errorLabel = i18n.t(i18n.l.swap.actions.error);
```

Comment:
in one of the tickets, I think Christian decided that we don't want to have a button label with just Error on it